Components/Queues/AsyncQueueConsumer.php

```diff
@@ -4,90 +4,164 @@
 
 namespace Smartbox\Integration\FrameworkBundle\Components\Queues;
 
-use Psr\Log\LoggerAwareInterface;
-use Psr\Log\LoggerAwareTrait;
-use Smartbox\Integration\FrameworkBundle\Components\Queues\Handler\AmqpQueueHandler;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\ConsumerInterface;
-use Smartbox\Integration\FrameworkBundle\Core\Consumers\IsStopableConsumer;
+use PhpAmqpLib\Message\AMQPMessage;
+use Smartbox\CoreBundle\Type\SerializableInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\AsyncQueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Components\Queues\Drivers\QueueDriverInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Consumers\AbstractAsyncConsumer;
+use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointFactory;
 use Smartbox\Integration\FrameworkBundle\Core\Endpoints\EndpointInterface;
+use Smartbox\Integration\FrameworkBundle\Core\Messages\MessageInterface;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSerializer;
 use Smartbox\Integration\FrameworkBundle\DependencyInjection\Traits\UsesSmartesbHelper;
 use Smartbox\Integration\FrameworkBundle\Exceptions\Handler\UsesExceptionHandlerTrait;
-use Smartbox\Integration\FrameworkBundle\Service;
 
-class AsyncQueueConsumer extends Service implements ConsumerInterface, LoggerAwareInterface
+/**
+ * Class AsyncQueueConsumer
+ */
+class AsyncQueueConsumer extends AbstractAsyncConsumer
 {
-    use IsStopableConsumer;
     use UsesSmartesbHelper;
     use UsesSerializer;
-    use LoggerAwareTrait;
     use UsesExceptionHandlerTrait;
 
     /**
-     * @var QueueManager
+     * Consumer identifier name.
      */
-    private $manager;
+    const CONSUMER_TAG = 'amqp-consumer-%s-%s';
 
     /**
-     * @var string
+     * {@inheritdoc}
      */
-    private $format;
+    protected function initialize(EndpointInterface $endpoint)
+    {
+        if (!($driver = $this->getQueueDriver($endpoint))->isConnected()) {
+            $driver->connect();
+        }
+    }
 
     /**
-     * AsyncQueueConsumer constructor.
+     * Get the driver responsible to establish a communication with the broker.
      *
-     * @param QueueManager $manager
-     * @param string       $format
+     * @param EndpointInterface $endpoint
+     * @return QueueDriverInterface
+     */
+    protected function getQueueDriver(EndpointInterface $endpoint): QueueDriverInterface
```

Comment:
On the method definition, the expect driver return is an instance of `AsyncQueueDriverInterface` and not `QueueDriverInterface`

Reply:
Yes, I agree. I changed the method to this way.

Reply:
Done.